controllers/argocd/application_controller.go

```diff
@@ -100,6 +100,11 @@ func createUnstructuredApplication(app *v1alpha1.Application) (result *unstructu
 		argoApp.Spec.Project = "default"
 	}
 
+	// application destination can't have both name and server defined
+	if argoApp.Spec.Destination.Name != "" && argoApp.Spec.Destination.Server != "" {
```

Comment:
I made this small change here. It has nothing about this PR. I can file a new PR against this if you think it should not be here.